cli/cmd/upgrade/upgrade.go

```diff
@@ -0,0 +1,453 @@
+	if c.kubernetes == nil {
+		restConfig, err := settings.RESTClientGetter().ToRESTConfig()
+		if err != nil {
+			c.UI.Output("Retrieving Kubernetes auth: %v", err, terminal.WithErrorStyle())
+			return 1
+		}
+		c.kubernetes, err = kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			c.UI.Output("Initializing Kubernetes client: %v", err, terminal.WithErrorStyle())
+			return 1
+		}
+	}
+
+	c.UI.Output("Pre-Upgrade Checks", terminal.WithHeaderStyle())
+
+	// Note the logic here, common's CheckForInstallations function returns an error if
+	// the release is not found. In `upgrade` we should indeed error if a user doesn't currently have a release.
+	var foundNamespace string
+	if name, ns, err := common.CheckForInstallations(settings, uiLogger); err != nil {
+		c.UI.Output("could not find existing Consul installation - run `consul-k8s install`")
+		return 1
+	} else {
+		c.UI.Output("Existing installation found.", terminal.WithSuccessStyle())
+		c.UI.Output("Name: %s", name, terminal.WithInfoStyle())
+		c.UI.Output("Namespace: %s", ns, terminal.WithInfoStyle())
+
+		foundNamespace = ns
+	}
+
+	// Handle preset, value files, and set values logic.
+	vals, err := c.mergeValuesFlagsWithPrecedence(settings)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+	valuesYaml, err := yaml.Marshal(vals)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Print out the upgrade summary.
+	if !c.flagAutoApprove {
+		c.UI.Output("Consul Upgrade Summary", terminal.WithHeaderStyle())
+		c.UI.Output("Installation name: %s", common.DefaultReleaseName, terminal.WithInfoStyle())
+		c.UI.Output("Namespace: %s", foundNamespace, terminal.WithInfoStyle())
+
+		if len(vals) == 0 {
+			c.UI.Output("Overrides: "+string(valuesYaml), terminal.WithInfoStyle())
+		} else {
+			c.UI.Output("Overrides:"+"\n"+string(valuesYaml), terminal.WithInfoStyle())
+		}
+	}
+
+	// Without informing the user, default global.name to consul if it hasn't been set already. We don't allow setting
+	// the release name, and since that is hardcoded to "consul", setting global.name to "consul" makes it so resources
+	// aren't double prefixed with "consul-consul-...".
+	vals = install.MergeMaps(config.Convert(config.GlobalNameConsul), vals)
+
+	if !c.flagAutoApprove && !c.flagDryRun {
+		confirmation, err := c.UI.Input(&terminal.Input{
+			Prompt: "Proceed with upgrade? (y/N)",
+			Style:  terminal.InfoStyle,
+			Secret: false,
+		})
+
+		if err != nil {
+			c.UI.Output(err.Error(), terminal.WithErrorStyle())
+			return 1
+		}
+		if common.Abort(confirmation) {
+			c.UI.Output("Upgrade aborted. To learn how to customize your upgrade, run:\nconsul-k8s upgrade --help", terminal.WithInfoStyle())
+			return 1
+		}
+	}
+
+	if !c.flagDryRun {
+		c.UI.Output("Running Upgrade", terminal.WithHeaderStyle())
+	} else {
+		c.UI.Output("Performing Dry Upgrade", terminal.WithHeaderStyle())
```

Comment:
```suggestion
		c.UI.Output("Performing Dry Run Upgrade", terminal.WithHeaderStyle())
```

Reply:
Lol, much better wording. I didn't notice that.